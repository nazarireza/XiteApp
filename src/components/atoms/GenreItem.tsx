import React, {memo} from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';
import {colors, typography} from '../../assets';

interface GenreItemProps {
  name: string;
  selected?: boolean;
}

export const GenreItem: React.FC<TouchableOpacityProps & GenreItemProps> = memo(
  ({name, selected, style, ...rest}) => {
    return (
      <TouchableOpacity
        activeOpacity={0.7}
        style={[styles.container, style, selected && styles.selectedContainer]}
        {...rest}>
        <Text style={[typography.title, selected && styles.selectedText]}>
          {name}
        </Text>
      </TouchableOpacity>
    );
  },
);

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 8,
    borderWidth: 1,
    borderColor: colors.background.card,
    borderRadius: 5,
  },
  selectedContainer: {
    backgroundColor: colors.primary,
  },
  selectedText: {
    color: colors.text.primary,
  },
});
