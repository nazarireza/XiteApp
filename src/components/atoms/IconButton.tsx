import React, {memo} from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';

export const IconButton: React.FC<TouchableOpacityProps> = memo(
  ({activeOpacity = 0.6, style, disabled, ...rest}) => {
    return (
      <TouchableOpacity
        style={[styles.container, disabled && styles.disabledContainer, style]}
        {...{activeOpacity, disabled}}
        {...rest}
      />
    );
  },
);

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  disabledContainer: {
    opacity: 0.3,
  },
});
