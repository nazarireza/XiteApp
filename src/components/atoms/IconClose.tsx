import React, {memo} from 'react';
import Svg, {Path} from 'react-native-svg';
import {colors} from '../../assets';
import {IconBaseProps} from '.';

export const IconClose: React.FC<IconBaseProps> = memo(
  ({color = colors.text.primary, size = 24}) => {
    return (
      <Svg width={size} height={size} viewBox="0 0 24 24">
        <Path
          fill={color}
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </Svg>
    );
  },
);
