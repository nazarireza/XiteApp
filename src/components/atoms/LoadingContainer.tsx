import React, {memo} from 'react';
import {View, ActivityIndicator, StyleSheet, ViewProps} from 'react-native';
import {colors} from '../../assets';

interface LoadingContainerProps {
  isBusy?: boolean;
}

export const LoadingContainer: React.FC<ViewProps & LoadingContainerProps> =
  memo(({isBusy = false, style, children, ...rest}) => {
    return (
      <View style={[styles.container, style]} {...rest}>
        {children}
        {isBusy && (
          <View style={styles.indicatorContainer}>
            <ActivityIndicator size="large" color={colors.primary} />
          </View>
        )}
      </View>
    );
  });

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  indicatorContainer: {
    ...StyleSheet.absoluteFillObject,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
