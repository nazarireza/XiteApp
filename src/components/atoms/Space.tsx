import React, {memo} from 'react';
import {View} from 'react-native';

interface SpaceProps {
  size?: number;
}

export const Space: React.FC<SpaceProps> = memo(({size = 16}) => {
  return <View style={{height: size, width: size}} />;
});
