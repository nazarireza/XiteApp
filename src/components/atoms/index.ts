export * from './IconClose';
export * from './IconSearch';
export * from './IconBack';
export * from './Space';
export * from './IconButton';
export * from './IconFilter';
export * from './GenreItem';
export * from './VideoItem';
