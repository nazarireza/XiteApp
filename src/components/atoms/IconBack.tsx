import React, {memo} from 'react';
import Svg, {Path} from 'react-native-svg';
import { colors } from '../../assets';
import {IconBaseProps} from '../../types/components';

export const IconBack: React.FC<IconBaseProps> = memo(
  ({color = colors.text.primary, size = 24}) => {
    return (
      <Svg width={size} height={size} viewBox="0 0 24 24">
        <Path
          fill={color}
          d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"
        />
      </Svg>
    );
  },
);
