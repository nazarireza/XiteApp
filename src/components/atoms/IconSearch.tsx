import React, {memo} from 'react';
import Svg, {Path} from 'react-native-svg';
import {colors} from '../../assets';
import {IconBaseProps} from '../../types/components';

export const IconSearch: React.FC<IconBaseProps> = memo(
  ({color = colors.text.primary, size = 24}) => {
    return (
      <Svg width={size} height={size} viewBox="0 0 24 24">
        <Path
          fill={color}
          fillRule="nonzero"
          d="M16.294 14.579h-.903l-.32-.31a7.4 7.4 0 001.795-4.836 7.433 7.433 0 10-7.433 7.433c1.84 0 3.533-.675 4.837-1.796l.309.32v.904L20.296 22 22 20.296l-5.706-5.717zm-6.861 0a5.139 5.139 0 01-5.146-5.146 5.138 5.138 0 015.146-5.146 5.139 5.139 0 015.146 5.146 5.14 5.14 0 01-5.146 5.146z"
        />
      </Svg>
    );
  },
);
