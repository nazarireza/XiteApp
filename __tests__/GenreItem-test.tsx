import 'react-native';
import React from 'react';
import {GenreItem} from '../src/components/atoms/GenreItem';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const wrapper = renderer.create(<GenreItem name="Genre1" />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with selected prop', () => {
  const wrapper = renderer.create(<GenreItem name="Genre1" selected />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
