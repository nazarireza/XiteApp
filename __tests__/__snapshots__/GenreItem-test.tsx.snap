// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#2E2E3F",
      "borderRadius": 5,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#F1F2F4",
          "fontFamily": "Montserrat",
          "fontSize": 15,
        },
        undefined,
      ]
    }
  >
    Genre1
  </Text>
</View>
`;

exports[`renders correctly with selected prop 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0078EE",
      "borderColor": "#2E2E3F",
      "borderRadius": 5,
      "borderWidth": 1,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#F1F2F4",
          "fontFamily": "Montserrat",
          "fontSize": 15,
        },
        Object {
          "color": "#F1F2F4",
        },
      ]
    }
  >
    Genre1
  </Text>
</View>
`;
